scripts/visualise_results.py

```diff
@@ -8,35 +8,36 @@
 PERCENTILES_TO_GRAPH = (50, 90, 95, 99)
 
 
-def plot_data(df, number_of_days_to_plot):
-    plt.style.use('fast')
+class GraphGenerationFailed(Exception):
+    pass
 
-    if (
-        number_of_days_to_plot and number_of_days_to_plot <= 45
-    ):  # To make the chart still easily digestible
-        df.plot.line(marker="o", markersize=8)
-        plt.grid(True, axis="both", alpha=0.3)
-    else:
-        df.plot.line()
 
-    plt.margins(0.03, 0.07)
-    plt.legend(frameon=True, prop={"size": 17})
-    plt.xticks(df.index, df["DATE"], size="small", rotation=90)
-    plt.yticks(size="small")
-    plt.ylabel("Average Response Time (ms)")
-    plt.xlabel("Run Date (YYYY-MM-DD)", labelpad=13)
+def plot_data(df, number_of_days_to_plot):
+    try:
+        plt.style.use('fast')
 
-    plt.savefig('performance_graph.png', bbox_inches="tight")
-    print("Graph saved as performance_graph.png")
+        if (
+            number_of_days_to_plot and number_of_days_to_plot <= 45
+        ):  # To make the chart still easily digestible
+            df.plot.line(marker="o", markersize=8)
+            plt.grid(True, axis="both", alpha=0.3)
+        else:
+            df.plot.line()
```

Comment:
Another quick question! I noticed that this condition is missing test coverage, is this one of the bits that can't be tested?

Reply:
This is one that needs to be looked at separately, i think we will raise a card to look at refactoring some of the benchmark scripts now that we have test coverage. Not sure that we ever use this else anymore.

Reply:
That makes sense, thanks 👍